yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -94,6 +98,14 @@ private[yarn] class YarnAllocator(
   // Additional memory overhead.
   protected val memoryOverhead: Int = sparkConf.getInt("spark.yarn.executor.memoryOverhead",
     math.max((MEMORY_OVERHEAD_FACTOR * executorMemory).toInt, MEMORY_OVERHEAD_MIN))
+
+  // Make the maximum executor failure check to be relative with respect to duration
+  private val relativeMaxExecutorFailureCheck = 
```

Comment:
My point was that the size of the window might not need to vary.  Are there examples that come to mind of scenarios where the ideal window size is widely different?